In [1]:
import pandas as pd
import numpy as np
import seaborn as sns

In [2]:
salary_test = pd.read_csv("SalaryData_Test.csv")
salary_test.head()

age   workclass      education  educationno        maritalstatus  \
0   25     Private           11th            7        Never-married   
1   38     Private        HS-grad            9   Married-civ-spouse   
2   28   Local-gov     Assoc-acdm           12   Married-civ-spouse   
3   44     Private   Some-college           10   Married-civ-spouse   
4   34     Private           10th            6        Never-married   

           occupation    relationship    race    sex  capitalgain  \
0   Machine-op-inspct       Own-child   Black   Male            0   
1     Farming-fishing         Husband   White   Male            0   
2     Protective-serv         Husband   White   Male            0   
3   Machine-op-inspct         Husband   Black   Male         7688   
4       Other-service   Not-in-family   White   Male            0   

   capitalloss  hoursperweek          native  Salary  
0            0            40   United-States   <=50K  
1            0            50   United-States   <=50K  
2            0            40   United-States    >50K  
3            0            40   United-States    >50K  
4            0            30   United-States   <=50K

In [3]:
salary_test.shape

(15060, 14)

In [4]:
salary_test.isna().sum()

age              0
workclass        0
education        0
educationno      0
maritalstatus    0
occupation       0
relationship     0
race             0
sex              0
capitalgain      0
capitalloss      0
hoursperweek     0
native           0
Salary           0
dtype: int64

In [5]:
salary_test.loc[salary_test['Salary'] == ' >50K', 'Salary'] = 1
salary_test.loc[salary_test['Salary'] == ' <=50K', 'Salary'] = 0

In [6]:
salary_test

age      workclass      education  educationno        maritalstatus  \
0       25        Private           11th            7        Never-married   
1       38        Private        HS-grad            9   Married-civ-spouse   
2       28      Local-gov     Assoc-acdm           12   Married-civ-spouse   
3       44        Private   Some-college           10   Married-civ-spouse   
4       34        Private           10th            6        Never-married   
...    ...            ...            ...          ...                  ...   
15055   33        Private      Bachelors           13        Never-married   
15056   39        Private      Bachelors           13             Divorced   
15057   38        Private      Bachelors           13   Married-civ-spouse   
15058   44        Private      Bachelors           13             Divorced   
15059   35   Self-emp-inc      Bachelors           13   Married-civ-spouse   

               occupation    relationship                 race      sex  \
0       Machine-op-inspct       Own-child                Black     Male   
1         Farming-fishing         Husband                White     Male   
2         Protective-serv         Husband                White     Male   
3       Machine-op-inspct         Husband                Black     Male   
4           Other-service   Not-in-family                White     Male   
...                   ...             ...                  ...      ...   
15055      Prof-specialty       Own-child                White     Male   
15056      Prof-specialty   Not-in-family                White   Female   
15057      Prof-specialty         Husband                White     Male   
15058        Adm-clerical       Own-child   Asian-Pac-Islander     Male   
15059     Exec-managerial         Husband                White     Male   

       capitalgain  capitalloss  hoursperweek          native Salary  
0                0            0            40   United-States      0  
1                0            0            50   United-States      0  
2                0            0            40   United-States      1  
3             7688            0            40   United-States      1  
4                0            0            30   United-States      0  
...            ...          ...           ...             ...    ...  
15055            0            0            40   United-States      0  
15056            0            0            36   United-States      0  
15057            0            0            50   United-States      0  
15058         5455            0            40   United-States      0  
15059            0            0            60   United-States      1  

[15060 rows x 14 columns]

In [7]:
from sklearn.preprocessing import LabelEncoder
labelencoder_test = LabelEncoder()
salary_test=salary_test.apply(LabelEncoder().fit_transform)

In [8]:
X_test = salary_test.drop(['Salary'],axis = 1)
Y_test = salary_test['Salary']

In [9]:
X_test

age  workclass  education  educationno  maritalstatus  occupation  \
0        8          2          1            6              4           6   
1       21          2         11            8              2           4   
2       11          1          7           11              2          10   
3       27          2         15            9              2           6   
4       17          2          0            5              4           7   
...    ...        ...        ...          ...            ...         ...   
15055   16          2          9           12              4           9   
15056   22          2          9           12              0           9   
15057   21          2          9           12              2           9   
15058   27          2          9           12              0           0   
15059   18          3          9           12              2           3   

       relationship  race  sex  capitalgain  capitalloss  hoursperweek  native  
0                 3     2    1            0            0            39      37  
1                 0     4    1            0            0            49      37  
2                 0     4    1            0            0            39      37  
3                 0     2    1           87            0            39      37  
4                 1     4    1            0            0            29      37  
...             ...   ...  ...          ...          ...           ...     ...  
15055             3     4    1            0            0            39      37  
15056             1     4    0            0            0            35      37  
15057             0     4    1            0            0            49      37  
15058             3     1    1           73            0            39      37  
15059             0     4    1            0            0            59      37  

[15060 rows x 13 columns]

In [10]:
salary_train=pd.read_csv('SalaryData_Train.csv')
salary_train

age          workclass    education  educationno        maritalstatus  \
0       39          State-gov    Bachelors           13        Never-married   
1       50   Self-emp-not-inc    Bachelors           13   Married-civ-spouse   
2       38            Private      HS-grad            9             Divorced   
3       53            Private         11th            7   Married-civ-spouse   
4       28            Private    Bachelors           13   Married-civ-spouse   
...    ...                ...          ...          ...                  ...   
30156   27            Private   Assoc-acdm           12   Married-civ-spouse   
30157   40            Private      HS-grad            9   Married-civ-spouse   
30158   58            Private      HS-grad            9              Widowed   
30159   22            Private      HS-grad            9        Never-married   
30160   52       Self-emp-inc      HS-grad            9   Married-civ-spouse   

               occupation    relationship    race      sex  capitalgain  \
0            Adm-clerical   Not-in-family   White     Male         2174   
1         Exec-managerial         Husband   White     Male            0   
2       Handlers-cleaners   Not-in-family   White     Male            0   
3       Handlers-cleaners         Husband   Black     Male            0   
4          Prof-specialty            Wife   Black   Female            0   
...                   ...             ...     ...      ...          ...   
30156        Tech-support            Wife   White   Female            0   
30157   Machine-op-inspct         Husband   White     Male            0   
30158        Adm-clerical       Unmarried   White   Female            0   
30159        Adm-clerical       Own-child   White     Male            0   
30160     Exec-managerial            Wife   White   Female        15024   

       capitalloss  hoursperweek          native  Salary  
0                0            40   United-States   <=50K  
1                0            13   United-States   <=50K  
2                0            40   United-States   <=50K  
3                0            40   United-States   <=50K  
4                0            40            Cuba   <=50K  
...            ...           ...             ...     ...  
30156            0            38   United-States   <=50K  
30157            0            40   United-States    >50K  
30158            0            40   United-States   <=50K  
30159            0            20   United-States   <=50K  
30160            0            40   United-States    >50K  

[30161 rows x 14 columns]

In [11]:
salary_train.shape

(30161, 14)

In [12]:
salary_train.isna().sum()

age              0
workclass        0
education        0
educationno      0
maritalstatus    0
occupation       0
relationship     0
race             0
sex              0
capitalgain      0
capitalloss      0
hoursperweek     0
native           0
Salary           0
dtype: int64

In [13]:
salary_train.loc[salary_train['Salary'] == ' >50K', 'Salary'] = 1
salary_train.loc[salary_train['Salary'] == ' <=50K', 'Salary'] = 0

In [14]:
salary_train

age          workclass    education  educationno        maritalstatus  \
0       39          State-gov    Bachelors           13        Never-married   
1       50   Self-emp-not-inc    Bachelors           13   Married-civ-spouse   
2       38            Private      HS-grad            9             Divorced   
3       53            Private         11th            7   Married-civ-spouse   
4       28            Private    Bachelors           13   Married-civ-spouse   
...    ...                ...          ...          ...                  ...   
30156   27            Private   Assoc-acdm           12   Married-civ-spouse   
30157   40            Private      HS-grad            9   Married-civ-spouse   
30158   58            Private      HS-grad            9              Widowed   
30159   22            Private      HS-grad            9        Never-married   
30160   52       Self-emp-inc      HS-grad            9   Married-civ-spouse   

               occupation    relationship    race      sex  capitalgain  \
0            Adm-clerical   Not-in-family   White     Male         2174   
1         Exec-managerial         Husband   White     Male            0   
2       Handlers-cleaners   Not-in-family   White     Male            0   
3       Handlers-cleaners         Husband   Black     Male            0   
4          Prof-specialty            Wife   Black   Female            0   
...                   ...             ...     ...      ...          ...   
30156        Tech-support            Wife   White   Female            0   
30157   Machine-op-inspct         Husband   White     Male            0   
30158        Adm-clerical       Unmarried   White   Female            0   
30159        Adm-clerical       Own-child   White     Male            0   
30160     Exec-managerial            Wife   White   Female        15024   

       capitalloss  hoursperweek          native Salary  
0                0            40   United-States      0  
1                0            13   United-States      0  
2                0            40   United-States      0  
3                0            40   United-States      0  
4                0            40            Cuba      0  
...            ...           ...             ...    ...  
30156            0            38   United-States      0  
30157            0            40   United-States      1  
30158            0            40   United-States      0  
30159            0            20   United-States      0  
30160            0            40   United-States      1  

[30161 rows x 14 columns]

In [15]:
labelencoder_train = LabelEncoder()
salary_train=salary_train.apply(LabelEncoder().fit_transform)

In [16]:
salary_train

age  workclass  education  educationno  maritalstatus  occupation  \
0       22          5          9           12              4           0   
1       33          4          9           12              2           3   
2       21          2         11            8              0           5   
3       36          2          1            6              2           5   
4       11          2          9           12              2           9   
...    ...        ...        ...          ...            ...         ...   
30156   10          2          7           11              2          12   
30157   23          2         11            8              2           6   
30158   41          2         11            8              6           0   
30159    5          2         11            8              4           0   
30160   35          3         11            8              2           3   

       relationship  race  sex  capitalgain  capitalloss  hoursperweek  \
0                 1     4    1           24            0            39   
1                 0     4    1            0            0            12   
2                 1     4    1            0            0            39   
3                 0     2    1            0            0            39   
4                 5     2    0            0            0            39   
...             ...   ...  ...          ...          ...           ...   
30156             5     4    0            0            0            37   
30157             0     4    1            0            0            39   
30158             4     4    0            0            0            39   
30159             3     4    1            0            0            19   
30160             5     4    0          107            0            39   

       native  Salary  
0          37       0  
1          37       0  
2          37       0  
3          37       0  
4           4       0  
...       ...     ...  
30156      37       0  
30157      37       1  
30158      37       0  
30159      37       0  
30160      37       1  

[30161 rows x 14 columns]

In [17]:
X_train = salary_train.drop(['Salary'],axis=1)
Y_train = salary_train['Salary']

In [18]:
X_train 

age  workclass  education  educationno  maritalstatus  occupation  \
0       22          5          9           12              4           0   
1       33          4          9           12              2           3   
2       21          2         11            8              0           5   
3       36          2          1            6              2           5   
4       11          2          9           12              2           9   
...    ...        ...        ...          ...            ...         ...   
30156   10          2          7           11              2          12   
30157   23          2         11            8              2           6   
30158   41          2         11            8              6           0   
30159    5          2         11            8              4           0   
30160   35          3         11            8              2           3   

       relationship  race  sex  capitalgain  capitalloss  hoursperweek  native  
0                 1     4    1           24            0            39      37  
1                 0     4    1            0            0            12      37  
2                 1     4    1            0            0            39      37  
3                 0     2    1            0            0            39      37  
4                 5     2    0            0            0            39       4  
...             ...   ...  ...          ...          ...           ...     ...  
30156             5     4    0            0            0            37      37  
30157             0     4    1            0            0            39      37  
30158             4     4    0            0            0            39      37  
30159             3     4    1            0            0            19      37  
30160             5     4    0          107            0            39      37  

[30161 rows x 13 columns]

# Build Model using SVC for linear

In [19]:
from sklearn import svm
from sklearn.svm import SVC
from sklearn.model_selection import GridSearchCV
from sklearn.metrics import classification_report
from sklearn.model_selection import train_test_split

In [20]:
# kernel = linear
model_linear = SVC(kernel = "linear")
model_linear.fit(X_train,Y_train)
pred_test_linear = model_linear.predict(X_test)

np.mean(pred_test_linear==Y_test)


0.8041832669322709

#### Using linear Kernel for gas_turbines dataset accuary for prediction model is 80.4%

# Build Model using SVC for Polynomial

In [21]:
# kernel = poly
model_poly = SVC(kernel = "poly")
model_poly.fit(X_train,Y_train)
pred_test_poly = model_poly.predict(X_test)
np.mean(pred_test_poly==Y_test) 

0.8195219123505976

#### Using Polynomial Kernel for gas_turbines dataset accuary for prediction model is 81.9%

# Build Model using SVC for Radial basis function

In [22]:
# kernel = rbf
model_rbf = SVC(kernel = "rbf")
model_rbf.fit(X_train,Y_train)
pred_test_rbf = model_rbf.predict(X_test)

np.mean(pred_test_rbf==Y_test)


0.8118193891102258

#### Using Radial basis function Kernel for gas_turbines dataset accuary for prediction model is 81.18%

# Build Model using SVC for Sigmoid

In [23]:
#'sigmoid'
model_sig = SVC(kernel = "sigmoid")
model_sig.fit(X_train,Y_train)
pred_test_sig = model_rbf.predict(X_test)

np.mean(pred_test_sig==Y_test) 

0.8118193891102258

#### Using Sigmoid Kernel for gas_turbines dataset accuary for prediction model is 81.18%